### PR TITLE
Update rmcp to 1.7.0

### PR DESCRIPTION
**Pull request by anp-oai**
WIll make it easier to uprev when the new draft spec is supported.

Also updates reqwest where needed for compatibility but doesn't update it everywhere since this is already a large diff.

### CODE REVIEW

File: codex-rs/rmcp-client/src/http_client_adapter.rs [hunk 1]
```diff
@@ -376,6 +393,58 @@ fn response_header(headers: &[HttpHeader], name: impl AsRef<str>) -> Option<Stri
...
+        if !name
+            .split_ascii_whitespace()
+            .last()?
+            .eq_ignore_ascii_case("scope")
+        {
+            return None;
+        }
+
+        let value = value.trim();
+        if let Some(quoted_value) = value.strip_prefix('"') {
```
Comment (bolinfest) [inline codex-rs/rmcp-client/src/http_client_adapter.rs]: Can/should we try deserializing as JSON into a string literal value using serde? Not sure what the escaping rules are for this value?

File: codex-rs/rmcp-client/src/http_client_adapter.rs [hunk 2]
```diff
@@ -407,3 +476,65 @@ fn sse_stream_from_body(
     }))
     .boxed()
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (bolinfest) [inline codex-rs/rmcp-client/src/http_client_adapter.rs]: I guess so long as we have test coverage, I don't feel that strongly about the implementation.

Though use `[path]` and put these in a separate file?

File: codex-rs/rmcp-client/src/rmcp_client.rs [hunk 3]
```diff
@@ -1037,7 +1034,7 @@ async fn create_oauth_transport_and_runtime(
     let manager = match oauth_state {
         OAuthState::Authorized(manager) => manager,
         OAuthState::Unauthorized(manager) => manager,
-        OAuthState::Session(_) | OAuthState::AuthorizedHttpClient(_) => {
+        _ => {
```
Comment (bolinfest) [inline codex-rs/rmcp-client/src/rmcp_client.rs:1037]: Should we continue to be explicit so we're forced to update this code if a new variant is introduced?

**Review by bolinfest — APPROVED** [review]
Thanks for doing this migration: it feels like there are a lot of ergonomic API improvements as part of this!

File: codex-rs/rmcp-client/src/rmcp_client.rs (hunk 3 above)
Comment (anp-oai) [reply codex-rs/rmcp-client/src/rmcp_client.rs:1037]: IMO maybe yes, but the [rmcp update past 1.x](https://github.com/modelcontextprotocol/rust-sdk/discussions/716) moved these enums to be `#[non_exhaustive]` by default. I think if we wanted we could allow the non-exhaustive lint and still do exhaustive matching. WDYT?

File: codex-rs/rmcp-client/src/http_client_adapter.rs (hunk 1 above)
Comment (anp-oai) [reply codex-rs/rmcp-client/src/http_client_adapter.rs]: I read the spec a bit more closely and it's not JSON, its a specific header field format. Pointed codex at the specs and asked for edge cases and more tests, think the new parser should be more robust. I'm a little surprised we don't already have a helper for this, but it seems that the oauth2 crate doesn't expose one and rmcp's is crate-internal.

File: codex-rs/rmcp-client/src/rmcp_client.rs (hunk 3 above)
Comment (bolinfest) [reply codex-rs/rmcp-client/src/rmcp_client.rs:1037]: I don't feel that strongly. I suppose it's low enough probability that we would get a new variant that merits updating this code?

Comment (anp-oai) [reply codex-rs/rmcp-client/src/rmcp_client.rs:1037]: Discussed offline, shipping as-is
